specification/portalservices/resource-manager/Microsoft.PortalServices/preview/2023-01-01-preview/extensions.json

```diff
@@ -0,0 +1,108 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2023-01-01-preview",
+    "title": "portal",
+    "description": "Allows creation and deletion of Azure portal extensions."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/providers/Microsoft.PortalServices/compilefile": {
+      "post": {
+        "tags": [
+          "PortalTenant"
+        ],
+        "operationId": "PortalTenant_CompileFile",
+        "x-ms-examples": {
+          "Compile single file using inline content": {
+            "$ref": "examples/portalTenantCompileFile.json"
+          }
+        },
+        "description": "compiles a file using inline content.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "portalTenantCompileFile",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/PortalTenantCompileFile"
+            },
+            "description": "The parameters required to compile a file."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns the runtime configuration of the file.",
+            "schema": {
+              "$ref": "#/definitions/PortalTenantCompileFileResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "produces": [
+          "application/json"
+        ],
+        "consumes": [
+          "application/json"
+        ]
+      }
+    }
+  },
+  "definitions": {
+    "PortalTenantCompileFileResult": {
+      "type": "object",
+      "description": "The runtime result of source compilation",
+      "additionalProperties": true
+    },
+    "PortalTenantCompileFile": {
+      "type": "object",
+      "description": "The contents of the file to compile.",
+      "properties": {
+        "contents": {
+          "type": "object",
```

Comment:
>object [](http://example.com/codeflow?start=19&length=6)

Do you intend these to be free form objects? Can these be defined with some particular schemas?

Reply:
'contents' is affectively the file contents of a code\source file. The portal has a json format, known as dx, which is compiled into runtime experiences for the portal.

'stringSource' is the contents of a resjson file. Resjson is publicly documented but can't be defined in swagger as its a free form json object.